nano/lib/asio.hpp

```diff
@@ -0,0 +1,34 @@
+#pragma once
+
+#include <nano/boost/asio.hpp>
+
+namespace nano
+{
+class shared_const_buffer
+{
+public:
+	using value_type = boost::asio::const_buffer;
+	using const_iterator = const boost::asio::const_buffer *;
+
+	explicit shared_const_buffer (std::vector<uint8_t> const & data);
+	explicit shared_const_buffer (uint8_t data);
+	explicit shared_const_buffer (std::string const & data);
+	explicit shared_const_buffer (std::vector<uint8_t> && data);
+	explicit shared_const_buffer (std::shared_ptr<std::vector<uint8_t>> const & data);
+
+	const boost::asio::const_buffer * begin () const;
+	const boost::asio::const_buffer * end () const;
+
+	size_t size () const;
+
+private:
+	std::shared_ptr<std::vector<uint8_t>> m_data;
+	boost::asio::const_buffer m_buffer;
+};
+
```

Comment:
Maybe add a static assert here to ensure it's compliant to asio's ConstBufferSequence concept, should it change in the future. My notes says something this should work:

`static_assert(boost::asio::is_const_buffer_sequence<shared_const_buffer<uint8_t>>::value, "Not ConstBufferSequence compliant");`

Reply:
good idea